compiler/rustc_codegen_ssa/src/traits/builder.rs

```diff
@@ -555,12 +555,19 @@ pub trait BuilderMethods<'a, 'tcx>:
     /// Called for `StorageDead`
     fn lifetime_end(&mut self, ptr: Self::Value, size: Size);
 
+    /// "Finally codegen the call"
+    ///
+    /// The typical case that an argument is None is during the codegen of intrinsics,
+    /// as they are "fake functions" that have no meaningful ABI.
```

Comment:
Also callng lang items uses a None FnAbi. The FnAbi and Instance passed here are only used for CFI. The caller is expected to have already handled the ABI lowering.

Reply:
It is also used for applying some call-site attributes like nounwind or noreturn, but yeah, most of the code in it is for CFI

Reply:
@bjorn3 Can you please take another look at the latest version to see if I accurately described the situation?